### DISCUSSION TITLE
Replace System.Net.Mail with MailKit

### DISCUSSION THREAD

**Discussion by Hypnodude** [oqtane/oqtane.framework#4482] [General] [answered]
Hi,

From everything that I've read, it appears that System.Net.Mail has been replaced with the MailKit SDK and that System.Mail.Net has been deprecated.
If this is so, how come Oqtane is developed using System.Net.Mail ?
Should it not be replaced with MailKit?

Thanx,
Alon

**Comment by Hypnodude** [comment]
I'm happy to do the change and submit as a PR if it will help.
Please let me know.

Thanx,
Alon

**Comment by sbwalker** [comment, marked as answer]
One of the overarching principles (https://www.oqtane.org/blog/!/20/oqtane-philosophy) in Oqtane is "Minimizing Dependencies" - the rationale being that the greater the number of dependencies, the more points of failure which need to be managed. Points of failure can be anything from basic defects, to compatibility issues with newer technology, to licensing issues, to security vulnerabilities (CVEs). Dependencies are one of the  largest challenges in terms of managing a large scale software product. Oqtane has done a good job adhering to this strategy... the table below contains metrics related to the number of dependencies of various open source CMS's (based on GitHub Dependency graphs):

   67 	Oqtane (https://github.com/oqtane/oqtane.framework/network/dependencies)
  569	Umbraco (https://github.com/umbraco/Umbraco-CMS/network/dependencies)
1,150	Joomla! (https://github.com/joomla/joomla-cms/network/dependencies)
1,439	Drupal (https://github.com/drupal/drupal/network/dependencies)
2,360	WordPress (https://github.com/WordPress/WordPress/network/dependencies)
2,914	Orchard Core (https://github.com/OrchardCMS/OrchardCore/network/dependencies) 
3,605	DNN Platform (https://github.com/dnnsoftware/Dnn.Platform/network/dependencies)
5,120	Strapi (https://github.com/strapi/strapi/network/dependencies)
7,503 	ABP Framework (https://github.com/abpframework/abp/network/dependencies)

This being said, Microsoft has made it clear that System.Net.Mail is no longer recommended and that developers are encouraged to use alternatives such as MailKit. So I have no objection to replacing System.Net.Mail with MailKit in Oqtane.

**Reply by ADefWebserver** [reply]
If you take out the "Microsoft" controlled dependencies (understandable because "Blazer" itself is a "Microsoft" dependency), the list is a lot smaller 👍🏾
